### PR TITLE
GH-5: Fixed handling of `null` values.

### PR DESCRIPTION
**Pull request by the-t-in-rtf**
See #5 for details.

(Also removes ellipses from all test descriptions.)

### CODE REVIEW

File: src/js/diff.js [hunk 1]
```diff
@@ -960,7 +960,13 @@ fluid.diff.arraysEqual = function (leftArray, rightArray) {
  *
  */
 fluid.diff.objectsEqual = function (leftObject, rightObject) {
-    if (typeof leftObject !== typeof rightObject) {
+    if (leftObject === null && rightObject === null) {
```
Comment (amb26) [inline src/js/diff.js]: You might find that the use of a function like "fluid.typeCode" can short-circuit a lot of the logic in this leg - https://github.com/fluid-project/infusion/blob/main/src/framework/core/js/Fluid.js#L401

File: src/js/diff.js [hunk 2]
```diff
@@ -960,7 +960,13 @@ fluid.diff.arraysEqual = function (leftArray, rightArray) {
...
+        return true;
+    }
+    else if ((leftObject === null && rightObject !== null) || (leftObject !== null && rightObject === null)) {
```
Comment (amb26) [inline src/js/diff.js]: A way I can like to express this is 
if (leftObject === null ^ rightObject === null)

File: src/js/diff.js [hunk 3]
```diff
@@ -939,7 +939,7 @@ fluid.diff.arraysEqual = function (leftArray, rightArray) {
             else if (leftValue === null && rightValue === null) {
                 return true;
             }
-            else if (typeof leftValue === "object") {
+            else if (typeof leftValue === "object" && leftValue !== null) {
```
Comment (amb26) [inline src/js/diff.js]: I think these tests now coincide with the behaviour of fluid.isPlainObject(leftValue, true)

File: src/js/diff.js (hunk 1 above)
Comment (amb26) [reply src/js/diff.js]: Perhaps not actually fluid.typeCode itself, mind, but a function *like* fluid.typeCode : P 
"Countries like Australia, such as New Zealand, Australia itself, South Africa, ...."

Comment (the-t-in-rtf) [reply src/js/diff.js]: I took a stab at it and also used it in the `leftValue` / `rightValue` comparison below.

File: src/js/diff.js (hunk 2 above)
Comment (the-t-in-rtf) [reply src/js/diff.js]: Done.

File: src/js/diff.js (hunk 3 above)
Comment (the-t-in-rtf) [reply src/js/diff.js]: Done, throughout.

**Comment by the-t-in-rtf** [comment]
Cheers.  I cut the full 1.0.1 release just now.

**Pull request merged**